### PR TITLE
Add source editor for ArtifactTemplates

### PR DESCRIPTION
**Pull request by balzer814**
Also-by: Philipp Meyer <meyer.github@gmail.com>
Signed-off-by: Lukas Balzer <balzer814_dev@web.de>

Add editor to:
   - Add new Files 
   - Upload existing files in the upload area
   - Edit the contents of the selected file in a simple text editor
   - Copy all sources to the 'files' directory in the winery-repository

grafik

### CODE REVIEW

**Comment by genie-winery** [comment]
Can one of the admins verify this patch?

**Comment by koppor** [comment]
ok to test

File: org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/apiData/ArtifactResourceApiData.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+public class ArtifactResourceApiData {
+
+	public String fileName;
+	public String content;
+
+	public ArtifactResourceApiData(String fileName) {
+		this.fileName = fileName;
+	}
+	public ArtifactResourceApiData() {
```
Comment (koppor) [inline org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/resources/apiData/ArtifactResourceApiData.java]: Please fix white space

- one before method
- no white space in the empty method

Is this constructor really required?

File: org.eclipse.winery.repository.ui/src/app/instance/artifactTemplates/artifactSource/artifactSource.component.ts [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    }
+
+    private handleRename() {
+        this.notify.success('Successfully renamed ' + this.selectedFile.name + ' to ' + this.renameFileName);
+        this.selectedFile = null;
+        this.renameFileName = null;
+        this.loadFiles();
+        this.invalidateEditor();
+        this.loading = false;
+
```
Comment (koppor) [inline org.eclipse.winery.repository.ui/src/app/instance/artifactTemplates/artifactSource/artifactSource.component.ts]: Please no empty final lines

File: org.eclipse.winery.repository.ui/src/app/instance/artifactTemplates/artifactSource/artifactSource.component.ts [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+import { InstanceService } from '../../instance.service';
+
+@Component({
+    templateUrl: 'artifactSource.component.html',
+    styleUrls: [
+        'artifactSource.component.css'
+    ],
+    providers: [
+        ArtifactSourceService
+        // FilesService
```
Comment (koppor) [inline org.eclipse.winery.repository.ui/src/app/instance/artifactTemplates/artifactSource/artifactSource.component.ts]: Why this disabled thing? Can this be removed?

File: org.eclipse.winery.repository.ui/src/app/instance/artifactTemplates/artifactSource/artifactSource.component.ts [hunk 4]
```diff
@@ -0,0 +1,231 @@
+/**
+ * Copyright (c) 2017 University of Stuttgart.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * and the Apache License 2.0 which both accompany this distribution,
+ * and are available at http://www.eclipse.org/legal/epl-v20.html
+ * and http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Contributors:
+ *     Philipp Meyer - initial API and implementation
```
Comment (koppor) [inline org.eclipse.winery.repository.ui/src/app/instance/artifactTemplates/artifactSource/artifactSource.component.ts]: This is a new class. Please do not list contributors. They are not required at EPLv2 any more. See https://dev.eclipse.org/mhonarc/lists/winery-dev/msg00053.html

File: org.eclipse.winery.repository.ui/src/app/instance/artifactTemplates/artifactSource/artifactSource.component.html [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+/**
+ * Copyright (c) 2017 University of Stuttgart.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * and the Apache License 2.0 which both accompany this distribution,
+ * and are available at http://www.eclipse.org/legal/epl-v20.html
+ * and http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Contributors:
+ *     Philipp Meyer - initial API and implementation
```
Comment (koppor) [inline org.eclipse.winery.repository.ui/src/app/instance/artifactTemplates/artifactSource/artifactSource.component.html]: See below - just remove the contributors section

File: org.eclipse.winery.repository.ui/src/app/instance/artifactTemplates/artifactSource/artifactSource.component.css [hunk 6]
```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2017 University of Stuttgart.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * and the Apache License 2.0 which both accompany this distribution,
+ * and are available at http://www.eclipse.org/legal/epl-v20.html
+ * and http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Contributors:
+ *     Philipp Meyer - initial API and implementation
```
Comment (koppor) [inline org.eclipse.winery.repository.ui/src/app/instance/artifactTemplates/artifactSource/artifactSource.component.css]: See below

**Comment by koppor** [comment]
Please ensure that you went through the checklist (https://github.com/eclipse/winery/blob/master/.github/PULL_REQUEST_TEMPLATE.md)

I think, the current TODO list is as follows:

- [ ] Ensure that the commit message is [a good commit message](https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [ ] Change in CHANGELOG.md described
- [ ] <s>Tests created for changes</s>
- [x] Screenshots added (for UI changes)

The requirements on a good commit message include

> Start the line with "Fix", "Add", "Change" instead of "Fixed", "Added", "Changed".

**Comment by koppor** [comment]
Less than 1000 LOC, so I can directly merge. Thank you for the contribution!

**Pull request merged**